src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -77,17 +89,19 @@ public LocalDate getEndDate() {
         }
         ArrayList<Task> tasksToAdd = new ArrayList<>();
         while (currentDate.isBefore(this.endDate) || currentDate.isEqual(this.endDate)) {
-            LocalDateTime localDateTime = LocalDateTime.of(currentDate, getStartTime());
-            String dateTime = localDateTime.format(DateTime.FORMATTER);
-            DateTime eventDateTime = new DateTime(dateTime);
-            Task taskToAdd = new Task(title, eventDateTime, description, new Type("lesson"), new HashSet<>());
-            tasksToAdd.add(taskToAdd);
+            LocalDateTime localStartDateTime = LocalDateTime.of(currentDate, getStartTime());
+            LocalDateTime localEndDateTime = LocalDateTime.of(currentDate, getStartTime());
+            String startDateTimeString = localStartDateTime.format(DateUtil.DATE_FORMATTER);
```

Comment:
I believe this is where you use parser util, code like this will be repeated, better to abstract it out